drivers/network/dd/dc21x4/net21x4.inf

```diff
@@ -0,0 +1,155 @@
+BusType = 5
+CopyFiles = DC_CopyFiles.NT
+AddReg = Common, DC_SD, DC_AUI_BNC, DC_10
+
+[DC21140.ndi.NT]
+Characteristics = 0x84
+BusType = 5
+CopyFiles = DC_CopyFiles.NT
+AddReg = Common, DC_SD, DC_10, DC_100
+
+[DC21143.ndi.NT]
+Characteristics = 0x84
+BusType = 5
+CopyFiles = DC_CopyFiles.NT
+AddReg = Common, DC_SD, DC_AUI_BNC, DC_10, DC_100
+
+[DC21145.ndi.NT]
+Characteristics = 0x84
+BusType = 5
+CopyFiles = DC_CopyFiles.NT
+AddReg = Common, DC_SD, DC_10, DC_HMR
+
+[DC_CopyFiles.NT]
+dc21x4.sys
+
+[DC21040.ndi.NT.Services]
+AddService = dc21x4, 2, DC_Service_Inst, DC_EventLog
+
+[DC21041.ndi.NT.Services]
+AddService = dc21x4, 2, DC_Service_Inst, DC_EventLog
+
+[DC21140.ndi.NT.Services]
+AddService = dc21x4, 2, DC_Service_Inst, DC_EventLog
+
+[DC21143.ndi.NT.Services]
+AddService = dc21x4, 2, DC_Service_Inst, DC_EventLog
+
+[DC21145.ndi.NT.Services]
+AddService = dc21x4, 2, DC_Service_Inst, DC_EventLog
+
+[Common]
+HKR, Ndi,            Service,    0, "dc21x4"
+HKR, Ndi\Interfaces, UpperRange, 0, "ndis5"
+HKR, Ndi\Interfaces, LowerRange, 0, "ethernet"
+
+HKR, Ndi\params\NetworkAddress, ParamDesc, 0, %NA%
+HKR, Ndi\params\NetworkAddress, type,      0, "edit"
+HKR, Ndi\params\NetworkAddress, LimitText, 0, "12"
+HKR, Ndi\params\NetworkAddress, UpperCase, 0, "1"
+HKR, Ndi\params\NetworkAddress, default,   0, " "
+HKR, Ndi\params\NetworkAddress, optional,  0, "1"
+
+[DC_SD]
+HKR, Ndi\params\SpeedDuplex,      ParamDesc, 0, %SD%
+HKR, Ndi\params\SpeedDuplex,      type,      0, "enum"
+HKR, Ndi\params\SpeedDuplex,      default,   0, "0"
+HKR, Ndi\params\SpeedDuplex\enum, "10",      0, "%Auto%"
+
+[DC_AUIBNC]
+HKR, Ndi\params\SpeedDuplex\enum, "1", 0, "AUI/BNC"
+
+[DC_AUI_BNC]
+HKR, Ndi\params\SpeedDuplex\enum, "1", 0, "BNC"
+HKR, Ndi\params\SpeedDuplex\enum, "2", 0, "AUI"
+
+[DC_10]
+HKR, Ndi\params\SpeedDuplex\enum, "0", 0, "10T HD"
+HKR, Ndi\params\SpeedDuplex\enum, "4", 0, "10T FD"
+
+[DC_100]
+HKR, Ndi\params\SpeedDuplex\enum, "3", 0, "100TX HD"
+HKR, Ndi\params\SpeedDuplex\enum, "5", 0, "100TX FD"
+HKR, Ndi\params\SpeedDuplex\enum, "6", 0, "100T4"
+HKR, Ndi\params\SpeedDuplex\enum, "7", 0, "100FX HD"
+HKR, Ndi\params\SpeedDuplex\enum, "8", 0, "100FX FD"
+
+[DC_HMR]
+HKR, Ndi\params\SpeedDuplex\enum, "9", 0, "HomePNA"
+
+[DC_Service_Inst]
+ServiceType   = 1
+StartType     = 3
+ErrorControl  = 1
+ServiceBinary = %12%\dc21x4.sys
+LoadOrderGroup = NDIS
+
+[DC_EventLog]
+AddReg = DC_EventLog_AddReg
+
+[DC_EventLog_AddReg]
+HKR, , EventMessageFile, 0x00020000, "%%SystemRoot%%\System32\netevent.dll"
+HKR, , TypesSupported,   0x00010001, 7
+
+[Strings]
+; Non-localizable
+ReactOS="ReactOS Team"
+IntelMfg="Intel"
+
+NA="Network Address"
+SD="Speed & Duplex"
+Auto="Auto"
+
+DC21040.DeviceDesc="Intel 21040 PCI Ethernet Adapter"
+DC21041.DeviceDesc="Intel 21041 PCI Ethernet Adapter"
+DC21140.DeviceDesc="Intel 21140 PCI Ethernet Adapter"
+DC21143.DeviceDesc="Intel 21143 PCI Ethernet Adapter"
+DC21145.DeviceDesc="Intel 21145 PCI Ethernet Adapter"
```

Comment:
Were they all manufactured by Intel? Maybe some of them actually by DECchip? Also I remember Windows 7 has a distinct entry (perhaps based on subsystem value) detecting an emulated adapter by Virtual PC.

Reply:
Found it: `PCI\VEN_1011&DEV_0009&SUBSYS_21140A00&REV_20` - detection string for the emulated adapter. (Source: https://community.osr.com/discussion/comment/242520/#Comment_242520)

Reply:
Intel bought DEC in 1997 :)